### ISSUE TITLE
[ISSUE] databricks_job resources are always updated even when no changes

### ISSUE THREAD

**Issue by jordanjennings** [databrickslabs/terraform-provider-databricks#323]
### Terraform Version
```
Terraform v0.13.2
+ provider registry.terraform.io/databrickslabs/databricks v0.2.5
```

### Affected Resource(s)
- databricks_job

### Terraform Configuration Files
```hcl
resource "databricks_job" "test" {

  name = "Test"
  timeout_seconds = 1800
  max_retries = 1
  max_concurrent_runs = 1

  new_cluster {
    num_workers = 1
    spark_version = "7.2.x-scala2.12"
    node_type_id = "Standard_DS3_v2"
  }

  notebook_task {
    notebook_path = "/jobs/test"
    base_parameters = {
      param1 = "2"
      param2 = "false"
    }
  }

  email_notifications {
    on_failure = ["redacted1@email.com", "redacted2@email.com", "redacted3@email.com"]
  }

  schedule {
    timezone_id = "America/New_York"
    quartz_cron_expression = "0 0 * ? * * *" # hourly
  }
}

```

### Expected Behavior
After the job is created, when there are no changes to the configuration terraform should not try to change it.

### Actual Behavior
After initial creation, on each run of `terraform plan` / `terraform apply` it says there are changes for the job. Example plan below:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # databricks_job.salesforce will be updated in-place
  ~ resource "databricks_job" "test" {
        id                        = "8761"
        library_cran              = []
        library_egg               = []
        library_jar               = []
        library_maven             = []
        library_pypi              = []
        library_whl               = []
        max_concurrent_runs       = 1
        max_retries               = 1
        min_retry_interval_millis = 0
        name                      = "Test"
        retry_on_timeout          = false
        timeout_seconds           = 1800

        email_notifications {
            no_alert_for_skipped_runs = false
            on_failure                = [
                "redacted1@email.com",
                "redacted2@email.com",
                "redacted3@email.com",
            ]
            on_start                  = []
            on_success                = []
        }

      ~ new_cluster {
            autotermination_minutes      = 0
            custom_tags                  = {}
            enable_elastic_disk          = true
            enable_local_disk_encryption = false
            node_type_id                 = "Standard_DS3_v2"
            num_workers                  = 1
          ~ spark_conf                   = {
              - "spark.databricks.delta.preview.enabled" = "true" -> null
            }
            spark_env_vars               = {}
            spark_version                = "7.2.x-scala2.12"
            ssh_public_keys              = []
        }

        notebook_task {
            base_parameters = {
                "param1" = "2"
                "param2"   = "false"
            }
            notebook_path   = "/jobs/test"
        }

      ~ schedule {
          - pause_status           = "UNPAUSED" -> null
            quartz_cron_expression = "0 0 * ? * * *"
            timezone_id            = "America/New_York"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

In the `new_cluster` block it is trying to set `spark_conf` to the default (which I did not explicitly configure), and in the `schedule` block it is trying to set the `pause_status` to the default.

### Steps to Reproduce
Create the job by running `terraform apply`, then immediately run `terraform apply` again.

### Important Factoids
Running in Azure Databricks

As a workaround I can set `spark_conf` and `pause_status` to their default values and then no changes are detected with `plan` / `apply`.

**Comment by stikkireddy** [comment]
hey @jordanjennings add 
```spark_conf                   = {
              - "spark.databricks.delta.preview.enabled" = "true" -> null
            }```
to your new cluster config to resolve this. It is something that databricks automatically adds to your spark conf and you do not have it configured.

**Comment by jordanjennings** [comment]
@stikkireddy Thanks for the reply, yes that's exactly what I did to work around the issue.

It's still surprising behavior and I would consider it a bug because the user _must_ specify a default value that is listed as optional or things will not work as expected. Minimally this should be documented, but ideally it should not require users to configure it (unless they genuinely want something other than the default).

**Comment by nfx** [comment]
@jordanjennings released in 0.2.7
